Automattic/woocommerce-payments
# Reporting: Fix the `View report` link URL on Total payment volume tile, within Payment activity widget

Fixes #8706 

#### Changes proposed in this Pull Request

The PR fixes the `View report` link on Total Payment volume tile to match the requirement as follows: 

* `Total payment volume` should show all types created over the last 7 days (or selected time filter) except for `Loan disbursement` and `Loan repayment`

#### Notes

* The existing advanced filter on the Transaction page, do have `Type is` and `Type is not` filters, but they do not allow selecting more than one types in it, in the current design. 
* Thanks to @shendy-a8c , who drew my attention to the default Disputes link that uses search term params to give a result comprising types - `needs_response` and `warning_needs_response` 
<img width="275" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/4162931/6ffc570d-05e7-4059-bf96-1cc2572b4b3e">

* For segregating transactions matching the above criteria, similar principle has been used :
  * On the server, the [exact search](https://github.com/Automattic/woocommerce-payments-server/blob/b053dfccfe5ba8bd1545c71394a501bcffce02f1/server/wp-content/rest-api-plugins/endpoints/wcpay/repository/class-transaction-repository.php#L43-L47) of `type` has been added in the search fields. 
  * On the client, the `View report` URL was edited to include the search terms. 
  * Some minor CSS changes were done so that the tags that appear in the search box do not warp outside the box. 

A more permanent solution to this would be to have `typeIs` and `typeIsNot` params accept arrays. But this may require higher effort to execute and to ensure backward compatibility. I found using the way similar to Disputes a better option for now.

#### Testing instructions
* On your local server instance, please check out branch `add/type-to-search-by-fields`
* On the local test client setup, make sure your account has sufficient transactions within the last 7 days, and at least one `Loan disbursement` and `Loan repayment` .  See if you can switch to `acct_1OuYn1QpUShfetUA` which has these ( and should remain for the next few days. Else create the loan transactions using the steps in the PdjTHR-31T-p2. 
* Browse to `Payments > Overview` and click the `View report` against `Total payment volume`
* On the transactions report page that lands, you should see all transactions except Loan disbursement and Loan repayment

#### Screenshots

On a wider screen
![image](https://github.com/Automattic/woocommerce-payments/assets/4162931/96f08228-eb56-4fa8-b6f6-8a2c3bc29662)

On a narrow screen
![image](https://github.com/Automattic/woocommerce-payments/assets/4162931/d9092292-251a-44b1-a3ca-4cf1bc5672d3)

 Without the CSS change, it would appear like this:
![image](https://github.com/Automattic/woocommerce-payments/assets/4162931/32134aef-5ab9-4153-8638-9c0264871dbb)


-------------------


**Post merge**
